src/hammers/image_deployer.py

```diff
@@ -109,16 +109,30 @@ def get_site_images(connection):
 
 
 def should_sync_image(image_connection, image_name, site_images, current):
-    if image_name in site_images:
+    matching_images = [
+        site_image_name for site_image_name in site_images
+        if site_image_name == image_name
+    ]
+
+    if len(matching_images) == 1:
         logging.debug(f"Image {image_name} already in site images.")
+
         image = image_connection.image.find_image(image_name)
         image_properties = image.properties
         logging.debug(f"Image properties: {image_properties}")
         image_current_value = image_properties.get("current", None)
         logging.debug(f"Image {image_name} current value: {image_current_value}")
+
         if image_current_value is not None and image_current_value == current:
             logging.debug(f"Image {image_name} is already current.")
             return False
+
+    if len(matching_images)> 1:
+        logging.warning(f"Multiple images found with name {image_name}. "
+                            "Aborting sync of this image. "
+                            "Manual intervention required.")
+        return False
+
     return True
```

Comment:
took me a second in the diff view, but this is the "0 matching images found" case?

Reply:
correct